create.rb

```diff
@@ -0,0 +1,90 @@
+require './list'
+require './book'
+require './rental'
+require './student'
+require './teacher'
+
+class Create
+  def self.create_student(people)
+    puts
+    print 'Age: '
+    age = gets.chomp.strip.to_i
+
+    while age < 4 || age > 120
+      print 'Enter correct age: '
+      age = gets.chomp.strip.to_i
+    end
+
+    print 'Name: '
+    name = gets.chomp.strip.capitalize
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp.strip.upcase
+
+    case permission
+    when 'Y' || 'y'
+      permission = true
+    when 'N' || 'n'
+      permission = false
+    end
+
+    people << Student.new(age, nil, name, parent_permission: permission)
+    puts 'Person created successfully'
+    puts
+  end
+
+  def self.create_teacher(people)
+    puts
+    print 'Age: '
+    age = gets.chomp.strip.to_i
+    while age < 18 || age > 120
+      print 'Enter correct age: '
+      age = gets.chomp.strip.to_i
+    end
+    print 'Name: '
+    name = gets.chomp.strip.capitalize
+    print 'Specialization: '
+    specialization = gets.chomp.strip
+    people << Teacher.new(age, specialization, name)
+    puts 'Person created successfully'
+  end
+
+  def self.create_book(books)
+    puts
+    print 'Title: '
+    title = gets.chomp.strip.capitalize
+    print 'Author: '
+    author = gets.chomp.strip.capitalize
+    books << Book.new(title, author)
+    puts 'Book created successfully'
+  end
+
+  def self.create_rental(people, books)
+    puts
+    puts 'Select a book from the following list by number'
+    List.list_books(books)
+    book_option = gets.chomp.to_i
+
```

Comment:
- [ ] it would be better to separate the code that requests and validates inputs from the users like:
```ruby 
    print 'Name: '
    name = gets.chomp.strip.capitalize # code to request input from user 
    print 'Has parent permission? [Y/N]: '
    permission = gets.chomp.strip.upcase

    case permission
    when 'Y' || 'y' # code to validate user input
      permission = true
    when 'N' || 'n'
      permission = false
    end

```
and put them in methods and then use the methods inside `create_rental`, `create_book`, `create_person` `list_book`...etc that way you will achieve the single responsibility responsible please try to do that for all the methods

Reply:
Hi @youmari ,

Thanks for the thorough feedback. I have implemented the user input and validation methods within a new InputHelper class according to your suggestion. Now, the create methods are using these input methods.

Best Regards,
K. Inoulen Shanongba